### ISSUE TITLE
What's the reasoning behind logging Validation/Login decisions?

### ISSUE THREAD

**Issue by mastacheata** [OWASP/ASVS#376]
What's the security benefit of requirements like this:
> 5.1.1 - Verify that server-side input validation failures result in request rejection and are logged.

I get why one would want to log exceptions and stuff that didn't work as expected, but why would one log things that were already handled by the application just the way they were intended to.
Logging when an authentication fails might give you a hint that an account was hacked, but why would I log when someone makes a mistake while filling out a form?
Especially the requirement for input validation seems completely unreasonable to me.
People filling out forms are mistyping so often and even if we have client-side validation there's still a ton of submissions that are caught as invalid only on the server.
There's no chance anyone would find anything suspicious in those logs at all. I even doubt a computer could properly filter those massive amounts of logging to find anything that's off.

So what's the security benefit of logging failed validations (and even more so what's the security benefit of logging SUCCESSFUL validation - that seems even more unlikely to yield any results whatsoever related to security when I look at  it)

**Comment by jmanico** [comment]
Logging failed validations is a significant benefit for intrusion detection.

**Comment by mastacheata** [comment]
I get that an IDS/IPS would require access to all user inputs, but without an automated system in place, how do I benefit from logging that?
There's gonna be thousands of entries a day if I only have a couple hundred users. Noone's gonna analyse that manually.
Unless you already have a system in place that also has set rules for defining when something is to be regarded as human error vs an attack, there's no way I can see that this would benefit anyone.

Shouldn't that rather be part of a requirement to have an IDS/IPS in place at Level 2 or 3?

**Comment by jmanico** [comment]
Many requirements can fit into multiple categories. I like this one as it and where it is. But I respect your opinion. I’m sorry we differ.

**Comment by tghosth** [comment]
Would it be worth restricting this to log server input validation failures **where the user must have bypassed the client side input validation**? That is a good, high fidelity signal that something is going on.

**Comment by mastacheata** [comment]
I would rephrase that to say something like:
"Log server side validation faults for rules that are enforced on both the client and server"
That's more generic than bypassing as it would also catch errors with the client-side rules.

I'm not sure if that satisfies Jim, though ;)
> I like this one as it and where it is. But I respect your opinion. I’m sorry we differ.
I'm glad we dffer :P Otherwise life would be pretty dull if everyone always had the same opinion.

I'll stick to the feeling that this rule as it currently stands isn't gonna benefit the security. It's too broad and producing too much overhead for anyone to manually look into the logs and detect any anomalies.

**Comment by danielcuthbert** [comment]
I like your rephrased approach @mastacheata and I think we need to re-order the top few so that they flow better. Ill do this now.
